src/components/implementation/sched/fprr/sched.c

```diff
@@ -0,0 +1,98 @@
+#include <cos_defkernel_api.h>
+#include <sched.h>
+#include <sl.h>
+#include "sched_info.h"
+
+int
+sched_thd_wakeup_intern(spdid_t c, thdid_t t)
+{
+	if (!c || !sched_childinfo_find(c)) return -1;
```

Comment:
This is an internal API. 
`sched_childinfo` is a data-structure that tracks the `cos_defcompinfo`, `sl_thd *` to the init thread (which is probably not necessary), and flags of that component. 
`sched_childinfo_find` returns that struct and this condition is making sure that only the child component (scheduling or non-sched (application)) of this scheduling component is making this invocation request. 

I don't think I need mutual exclusion around accessing the local `sched_childinfo` as the data structure being accessed is modified only `initialization` time. And these are calls made much later.

Reply:
Internal APIs are those that are not used from outside of the component.  Isn't this the *opposite* of that?  Your `_intern` functions seem like the functions that are *exported* as part of the `sinv` API.

Do you want to design the system so that it cannot be extended to allow dynamic components in the future?  What is the benefit of making this restriction?  I'd like a discussion of the actual trade-offs here, not only a statement about the *current* very limited system.  If you only design for what you have now, then we'll have a long stream of bugs, and people's wasted time debugging in the future.

Reply:
I'm sorry about not giving my full thought here, I should have documented it in the code and I'll.
`sched_childinfo_find()` is called here to get the internal data-structure of that spdid for consistency check, to ensure that the child of this scheduling component is making this call. 
If this call was made by a non-child component thread (the case I can think of is only a bug in some code), then I'll not have a `sl_thd *` in the scheduler for that client component, so cannot use `sl_lock` here until I know that this is a child component making this call.